server/rpc/gateway.go

```diff
@@ -0,0 +1,120 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/medibloc/panacea-oracle/server/service/datadeal"
+	"github.com/medibloc/panacea-oracle/server/service/key"
+	"github.com/medibloc/panacea-oracle/server/service/status"
+	"github.com/medibloc/panacea-oracle/service"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/backoff"
+)
+
+type gatewayServer struct {
+	grpcConn *grpc.ClientConn
+	service  service.Service
+}
+
+func NewGatewayServer(svc service.Service) *gatewayServer {
+	return &gatewayServer{
+		service: svc,
+	}
+}
+
+func (s *gatewayServer) Run() error {
+	log.Info("Running the API server")
+
+	if err := s.generateAndSetGrpcConnection(); err != nil {
+		return err
+	}
+
+	mux := runtime.NewServeMux()
+	if err := s.registerServiceHandler(mux); err != nil {
+		return err
+	}
+
+	return s.listenAndServe(mux)
+}
+
+func (s *gatewayServer) generateAndSetGrpcConnection() error {
```

Comment:
```suggestion
func (s *gatewayServer) createAndSetGrpcConnection() error {
```
Better to say `create` :)

Reply:
OK. I will change to `create...`. Thank you